cfme/intelligence/reports/schedules.py

```diff
@@ -198,49 +197,56 @@ def enabled(self):
             if item['Name'] == self.name:
                 return item['Active'] == 'True'
 
+    @property
+    def fill_dict(self):
+        return ({
+            "name": self.name,
+            "description": self.description,
+            "active": self.active,
+            "report_filter": self.report_filter,
+            "timer": self.timer,
+            "email": self.email,
+            "email_options": self.email_options,
+        })
+
 
 @attr.s
 class ScheduleCollection(BaseCollection):
 
     ENTITY = Schedule
 
-    def create(self, name=None, description=None, report_filter=None, active=None,
-               timer=None, from_email=None, emails=None, email_options=None):
-        schedule = self.instantiate(name, description, report_filter, active=active, timer=timer,
-            emails=emails, email_options=email_options)
-        view = navigate_to(self, "Add")
-
-        view.fill(
-            {
-                "name": name,
-                "description": description,
-                "active": active,
-                "filter1": report_filter[0],
-                "filter2": report_filter[1],
-                "filter3": report_filter[2],
-                "run": timer.get("run"),
-                "timer_hour": timer.get("run_hour"),
-                "timer_day": timer.get("run_day"),
-                "timer_week": timer.get("run_week"),
-                "timer_month": timer.get("run_month"),
-                "time_zone": timer.get("time_zone"),
-                "starting_date": timer.get("starting_date"),
-                "hour": timer.get("starting_hour"),
-                "minute": timer.get("starting_minute"),
-                "emails_send": bool(emails),
-                "from_email": from_email,
-                "emails": emails,
-                "send_if_empty": email_options.get("send_if_empty"),
-                "send_txt": email_options.get("send_txt"),
-                "send_csv": email_options.get("send_csv"),
-                "send_pdf": email_options.get("send_pdf"),
-            }
+    def create(
+        self,
+        name,
+        description,
+        report_filter,
+        active=True,
+        timer=None,
+        email=None,
+        email_options=None,
+        cancel=False,
+    ):
+        if email:
+            email["emails_send"] = True
+        schedule = self.instantiate(
+            name=name,
+            description=description,
+            active=active,
+            report_filter=report_filter,
+            email=email,
+            email_options=email_options
         )
 
+        view = navigate_to(self, "Add")
+        view.fill(schedule.fill_dict)
+        if cancel:
+            view.cancel_button.click()
         view.add_button.click()
```

Comment:
Again - if `cancel` was `True`, this will try to click add button after canceling the form.